File: 2490-circular-sentence/2490-circular-sentence.py
class Solution:
    def isCircularSentence(self, s: str) -> bool:
        if s[0] != s[-1]:
            return False
        
        n = len(s)
        
        for i in range(n - 1):
            if i and s[i] == ' ' and s[i - 1] != s[i + 1]:
                return False
            
        return True

File: 2490-circular-sentence/2490-circular-sentence.java
class Solution {
    public boolean isCircularSentence(String sentence) {
        String a[] = sentence.split(" ");
        final int n = a.length;
        
        if (a[0].charAt(0) != a[n - 1].charAt(a[n - 1].length() - 1)) {
            return false;
        }
        
        for (int i = 0; i < n - 1; ++i) {
            int next_i = i + 1;
            char last_idx_of_i = a[i].charAt(a[i].length() - 1);
            char head_idx_of_next_i = a[next_i].charAt(0);
            
            if (last_idx_of_i != head_idx_of_next_i) {
                return false;
            }
        }
        
        return true;
    }
}